xbmc/input/InputCodingTableBaiduPY.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+/*
+*      Copyright (C) 2005-2013 Team XBMC
+*      http://xbmc.org
+*
+*  This Program is free software; you can redistribute it and/or modify
+*  it under the terms of the GNU General Public License as published by
+*  the Free Software Foundation; either version 2, or (at your option)
+*  any later version.
+*
+*  This Program is distributed in the hope that it will be useful,
+*  but WITHOUT ANY WARRANTY; without even the implied warranty of
+*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+*  GNU General Public License for more details.
+*
+*  You should have received a copy of the GNU General Public License
+*  along with XBMC; see the file COPYING.  If not, see
+*  <http://www.gnu.org/licenses/>.
+*
+*/
+
+#include <deque>
+#include "InputCodingTable.h"
+#include "threads/Thread.h"
+
+class CInputCodingTableBaiduPY : public IInputCodingTable, public CThread
+{
+public:
+  CInputCodingTableBaiduPY();
+  virtual ~CInputCodingTableBaiduPY() {}
+
+  virtual bool GetWordListPage(const std::string& strCode, bool isFirstPage);
+  virtual void Process();
+
+  virtual std::shared_ptr<std::vector<std::wstring>> GetResponse(int response);
+private:
+  std::wstring UnicodeToWString(const std::string& unicode);
+  void HandleResponse(const std::string& strCode, const std::string& response);
+
+  std::string m_code;
+  int m_api_begin; // baidu api begin num
+  int m_api_end;   // baidu api end num
+  bool m_api_nomore;
+
+  int m_messageCounter;
+
+  std::deque<std::string> m_work;
+  std::map<int, std::shared_ptr<std::vector<std::wstring>>> m_responses;

```

Comment:
@Paxxi this line cause error under linux and osx, like:
`error: no template named 'map' in namespace 'std'`
how to solve this?